src/movement.h

```diff
@@ -90,9 +96,9 @@ class MoveEvents final : public BaseEvents
 		LuaScriptInterface scriptInterface;
 };
 
-using StepFunction = std::function<uint32_t(Creature* creature, Item* item, const Position& pos)>;
-using MoveFunction = std::function<uint32_t(Item* item, Item* tileItem, const Position& pos)>;
-using EquipFunction = std::function<uint32_t(MoveEvent* moveEvent, Player* player, Item* item, slots_t slot, bool boolean)>;
+typedef uint32_t(StepFunction)(Creature* creature, Item* item, const Position& pos);
```

Comment:
`using StepFunction = uint32_t(Creature* creature, Item* item, const Position& pos)`

Same can be done with the rest of callbacks.

Reply:
this needs some deeper look into, whenever I try to adjust this, I'm either stuck not beeing able to use those functions properly from the luascripts.cpp or it throws me tons of errors

Reply:
You're probably forgetting a `*` somewhere:
https://godbolt.org/z/re1qPL

Reply:
fixed in [c73ed97](https://github.com/otland/forgottenserver/pull/2558/commits/c73ed9783e0c33312326a790475cd4eee4f61ab9)